model/src/main/java/org/projectnessie/error/NessieUnsupportedMediaTypeException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.error;
+
+/** This exception is thrown when the request's media type is not accepted at the specified path. */
+public class NessieUnsupportedMediaTypeException extends NessieConflictException {
```

Comment:
Wonder whether this should actually be a subclass of `NessieServiceException`, which is a `RuntimeExcepiton`.
`NessieConflictException` is meant for "application/data errors", but unsupported media types feel like network/configuration issues. WDYT?

Reply:
You're completely right.  That was careless on my part. I see now that ​`NessieConflictException`'s javadoc says prominently that it corresponds with HTTP status 409, and that every subclass of `NCE` corresponds to HTTP status 409 (except the one I introduced here).  That should have been my hint to look for something else, like `NessieServiceException`.  I'll change that.

Reply:
Hmm, it seems `NessieServceException` is in clients/client, whereas `NessieUnsupportedMediaTypeException` is currently in model.

There are a bunch of ways to resolve this.  What do you think about making `NessieUnsupportedMediaTypeException` extend `NessieRuntimeException` instead, where the latter currently lives in model?

Reply:
Argh - yea - I always run into that exception-hierarchy trap...

Yea, extending `NessieRuntimeException` seems fine.

Reply:
I've pushed this as a separate commit, eyeballing precedent in other closed PRs, but let me know if you'd rather see it squashed.

Aside from rationalizing the inheritance hierarchy, this also makes NessieUnsupportedMediaTypeException's message more consistent with other Nessie 4XX exception messages by indirectly involving NessieError#getFullMessage.  I've updated the one test that was sensitive to the change to match.

Reply:
Yup - that looks good to me. But CI's failing now.

Reply:
I forgot to run spotless, my mistake.  It should be fully applied with the last force-push.

Reply:
+1 to extending `NessieRuntimeException`